web/server/entrypoint.go

```diff
@@ -398,7 +398,23 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 		robotOptions = append(robotOptions, robotimpl.WithFTDC())
 	}
 
-	myRobot, err := robotimpl.New(ctx, processedConfig, s.logger, robotOptions...)
+	// Create `minimalProcessedConfig`, a copy of `fullProcessedConfig`. Remove
+	// all components, services, remotes, modules, and processes from
+	// `minimalProcessedConfig`. Create new robot with `minimalProcessedConfig`
+	// and immediately start web service. We need the machine to be reachable
+	// through the web service ASAP, even if some resources take a long time to
+	// initially configure.
+	minimalProcessedConfig, err := fullProcessedConfig.CopyOnlyPublicFields()
+	if err != nil {
+		return err
+	}
+	minimalProcessedConfig.Components = nil
+	minimalProcessedConfig.Services = nil
+	minimalProcessedConfig.Remotes = nil
+	minimalProcessedConfig.Modules = nil
+	minimalProcessedConfig.Processes = nil
+
+	myRobot, err := robotimpl.New(ctx, minimalProcessedConfig, s.logger, robotOptions...)
```

Comment:
Is this the best way of achieving what we want or the most expedient?

I'm fine with this as-is. And I'm kind of fine never coming back to think about this. But the whole "robot owns the web server" feels backwards.

There would be less states to consider if we could start a web service and register robots with it. And there'd be a small API that describes:
* What state the robot is in (startup or running) and
* which APIs are available, e.g: 
  * just "GetMachineStatus" and maybe "ResourceNames"
  * but none of "SetPower"/other resource specific APIs

But in this PR we have 90-100 lines between this comment/robot.New and when the web service is started. That's a lot of lines to accidentally break our contract and add some blocking code.

Reply:
> Is this the best way of achieving what we want or the most expedient?

Perhaps not, and I understand your argument there. I've introduced a slightly different/simpler mechanic for controlling the "initializing" value, so that _might_ address some of your concerns here. I didn't go so far as starting a web service and registering robots with it (if I'm understanding what you're saying.)